lib/serve/index.ts

```diff
@@ -148,12 +149,15 @@ export class LisaServeApplicationStack extends Stack {
         });
 
         const subNets = config.subnetIds && config.vpcId ? vpc.subnetSelection?.subnets : vpc.vpc.isolatedSubnets.concat(vpc.vpc.privateSubnets);
-        subNets?.forEach((subnet) => {
-            litellmDbSg.connections.allowFrom(
-                Peer.ipv4(subnet.ipv4CidrBlock),
-                Port.tcp(config.restApiConfig.rdsConfig.dbPort),
-                'Allow REST API private subnets to communicate with LiteLLM database',
-            );
+        subNets?.filter((subnet) => !isSubnetPublic(subnet)).forEach(async (subnet) => {
+            const cidrRange = await getSubnetCidrRange(subnet.subnetId);
+            if (cidrRange){
+                litellmDbSg.connections.allowFrom(
+                    Peer.ipv4(cidrRange),
+                    Port.tcp(config.restApiConfig.rdsConfig.dbPort),
+                    'Allow REST API private subnets to communicate with LiteLLM database',
+                );
+            }
```

Comment:
This could potentially be an issue for race conditions. As discussed in dev sync lets change this to do the calls sequentially.